src/IIIFAuth2/IIIFAuth2.API.Tests/Integration/ProbeServiceTests.cs

```diff
@@ -0,0 +1,318 @@
+﻿using System.Net;
+using IIIF.Auth.V2;
+using IIIF.Serialisation;
+using IIIFAuth2.API.Data;
+using IIIFAuth2.API.Data.Entities;
+using IIIFAuth2.API.Tests.TestingInfrastructure;
+
+namespace IIIFAuth2.API.Tests.Integration;
+
+[Trait("Category", "Integration")]
+[Collection(DatabaseCollection.CollectionName)]
+public class ProbeServiceTests : IClassFixture<AuthWebApplicationFactory>
+{
+    private readonly HttpClient httpClient;
+    private readonly AuthServicesContext dbContext;
+
+    public ProbeServiceTests(AuthWebApplicationFactory factory, DatabaseFixture dbFixture)
+    {
+        dbContext = dbFixture.DbContext;
+        httpClient = factory
+            .WithConnectionString(dbFixture.ConnectionString)
+            .CreateClient();
+
+        dbFixture.CleanUp();
+    }
+    
+    [Fact]
+    public async Task GetProbeService_Returns400StatusProperty_IfRolesMissing()
+    {
+        // Arrange
+        const string path = "probe/99/2/assetname";
+            
+        // Act
+        var response = await httpClient.GetAsync(path);
+            
+        // Assert
+        response.StatusCode.Should().Be(HttpStatusCode.OK);
+        var authProbeResult = (await response.Content.ReadAsStreamAsync()).FromJsonStream<AuthProbeResult2>();
+        authProbeResult.Status.Should().Be(400);
```

Comment:
Would it make sense to change`.Be(400)` to  `.Be((int)HttpStatusCode.NotFound)`?

Reply:
Added "because" reasons for assertions. The probe-service-response `Status` property is an integer so feels clearer to keep it that way. Assertions are now in format `authProbeResult.Status.Should().Be(400, "BadRequest (400) expected");`